drivers/ethernet/eth_stm32_hal.c

```diff
@@ -0,0 +1,394 @@
+
+	k_sem_give(&dev_data->rx_int_sem);
+}
+
+static inline void __eth_stm32_hal_get_clock(struct device *dev)
+{
+	struct eth_stm32_hal_dev_data *dev_data = DEV_DATA(dev);
+	struct device *clk = device_get_binding(STM32_CLOCK_CONTROL_NAME);
+
+	__ASSERT_NO_MSG(clk);
+
+	dev_data->clock = clk;
+}
+
+static int eth_initialize(struct device *dev)
+{
+	struct eth_stm32_hal_dev_data *dev_data;
+	struct eth_stm32_hal_dev_cfg *cfg;
+
+	__ASSERT_NO_MSG(dev != NULL);
+
+	dev_data = DEV_DATA(dev);
+	cfg = DEV_CFG(dev);
+
+	__ASSERT_NO_MSG(dev_data != NULL);
+	__ASSERT_NO_MSG(cfg != NULL);
+
+	__eth_stm32_hal_get_clock(dev);
+
+	/* enable clock */
+	clock_control_on(dev_data->clock,
+		(clock_control_subsys_t *)&cfg->pclken);
+	clock_control_on(dev_data->clock,
+		(clock_control_subsys_t *)&cfg->pclken_tx);
+	clock_control_on(dev_data->clock,
+		(clock_control_subsys_t *)&cfg->pclken_rx);
+	clock_control_on(dev_data->clock,
+		(clock_control_subsys_t *)&cfg->pclken_ptp);
+
+	__ASSERT_NO_MSG(cfg->config_func != NULL);
+
+	cfg->config_func();
+
+	return 0;
+}
+
+#if defined(CONFIG_ETH_STM32_HAL_RANDOM_MAC)
+static void generate_mac(u8_t *mac_addr)
+{
+	u32_t entropy;
+
+	entropy = sys_rand32_get();
+
+	mac_addr[3] = entropy >> 8;
+	mac_addr[4] = entropy >> 16;
+	/* Locally administered, unicast */
+	mac_addr[5] = ((entropy >> 0) & 0xfc) | 0x02;
+}
+#endif
+
+static void eth0_iface_init(struct net_if *iface)
+{
+	struct device *dev;
+	struct eth_stm32_hal_dev_data *dev_data;
+	ETH_HandleTypeDef *heth;
+
+	__ASSERT_NO_MSG(iface != NULL);
+
+	dev = net_if_get_device(iface);
+	__ASSERT_NO_MSG(dev != NULL);
+
+	dev_data = DEV_DATA(dev);
+	__ASSERT_NO_MSG(dev_data != NULL);
+
+	heth = &dev_data->heth;
+
+	dev_data->iface = iface;
+
+	/* Initialize semaphores */
+	k_mutex_init(&dev_data->tx_mutex);
+	k_sem_init(&dev_data->rx_int_sem, 0, UINT_MAX);
+
+	/* Start interruption-poll thread */
+	k_thread_create(&dev_data->rx_thread, dev_data->rx_thread_stack,
+			K_THREAD_STACK_SIZEOF(dev_data->rx_thread_stack),
+			rx_thread, (void *) dev, NULL, NULL,
+			K_PRIO_COOP(CONFIG_ETH_STM32_HAL_RX_THREAD_PRIO),
+			0, K_NO_WAIT);
+
+#if defined(CONFIG_ETH_STM32_HAL_RANDOM_MAC)
+	generate_mac(dev_data->mac_addr);
+#endif
+
+	heth->Instance = ETH;
+	heth->Init.AutoNegotiation = ETH_AUTONEGOTIATION_ENABLE;
+	heth->Init.PhyAddress = 0;
+	heth->Init.MACAddr = dev_data->mac_addr;
+	heth->Init.RxMode = ETH_RXINTERRUPT_MODE;
+	heth->Init.ChecksumMode = ETH_CHECKSUM_BY_SOFTWARE;
+	heth->Init.MediaInterface = ETH_MEDIA_INTERFACE_RMII;
+
+	if (HAL_ETH_Init(heth) != HAL_OK) {
+		SYS_LOG_ERR("HAL_ETH_Init failed\n");
+	}
+
+	HAL_ETH_DMATxDescListInit(heth, dma_tx_desc_tab,
+		&dma_tx_buffer[0][0], ETH_TXBUFNB);
+	HAL_ETH_DMARxDescListInit(heth, dma_rx_desc_tab,
+		&dma_rx_buffer[0][0], ETH_RXBUFNB);
+
+	HAL_ETH_Start(heth);
+
+	/* Register Ethernet MAC Address with the upper layer */
+	net_if_set_link_addr(iface, dev_data->mac_addr,
+		sizeof(dev_data->mac_addr),
```

Comment:
Retried it and now it didn't complain, maybe it was just a warning that only was shown because there also were some errors at that time. But anyway, indentation fixed.